### PR TITLE
Fix build error -- unused var

### PR DESCRIPTION
**Pull request by beeender**

### CODE REVIEW

**Review by chrishajas — APPROVED** [review]
LGTM. Is there a flag we can add so these get reported in CI?

File: src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp [hunk 1]
```diff
@@ -1727,7 +1727,7 @@ CTranslatorExprToDXL::PdxlnDynamicForeignScan(
 	{
 		const CBitSet *bs = pps_reqd->SelectorIds(popDFS->ScanId());
 		CBitSetIter bsi(*bs);
-		for (ULONG ul = 0; bsi.Advance(); ul++)
+		for (; bsi.Advance();)
```
Comment (gpopt) [inline src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp:1730]: Can we use `while`?

**Review by gpopt — APPROVED** [review]
LGTM after comment is addressed.

**Comment by beeender** [comment]
> LGTM. Is there a flag we can add so these get reported in CI?

`-Werror,-Wunused-but-set-variable` this should do. It is on by default in the latest clang.

**Pull request merged**